pkg/clients/eventstreamsadminv1/eventstreamsadminv1.go

```diff
@@ -0,0 +1,234 @@
+		RetentionMs:        ibmc.Int64Value(in.RetentionMs),
+		CleanupPolicy:      reference.FromPtrValue(in.CleanupPolicy),
+		Configs:            Generatev1alpha1TopicConfigs(in.Configs),
+		ReplicaAssignments: GenerateReplicaAssignments(in.ReplicaAssignments),
+	}
+	return o, nil
+}
+
+// Generatev1alpha1TopicConfigs generates *v1alpha1.TopicConfigs from *arv1.TopicConfigs
+func Generatev1alpha1TopicConfigs(in *arv1.TopicConfigs) *v1alpha1.TopicConfigs {
+	if in == nil {
+		return nil
+	}
+	o := &v1alpha1.TopicConfigs{
+		CleanupPolicy:     reference.FromPtrValue(in.CleanupPolicy),
+		MinInsyncReplicas: reference.FromPtrValue(in.MinInsyncReplicas),
+		RetentionBytes:    reference.FromPtrValue(in.RetentionBytes),
+		RetentionMs:       reference.FromPtrValue(in.RetentionMs),
+		SegmentBytes:      reference.FromPtrValue(in.SegmentBytes),
+		SegmentIndexBytes: reference.FromPtrValue(in.SegmentIndexBytes),
+		SegmentMs:         reference.FromPtrValue(in.SegmentMs),
+	}
+	return o
+}
+
+// GenerateReplicaAssignments generates []v1alpha1.ReplicaAssignment from []arv1.ReplicaAssignment
+func GenerateReplicaAssignments(in []arv1.ReplicaAssignment) []v1alpha1.ReplicaAssignment {
+	if in == nil {
+		return nil
+	}
+	o := []v1alpha1.ReplicaAssignment{}
+	for _, m := range in {
+		item := v1alpha1.ReplicaAssignment{
+			ID:      ibmc.Int64Value(m.ID),
+			Brokers: GenerateBrokers(m.Brokers),
+		}
+		o = append(o, item)
+	}
+	return o
+}
+
+// GenerateBrokers generates *v1alpha1.ReplicaAssignmentBrokers from *arv1.ReplicaAssignmentBrokers
+func GenerateBrokers(in *arv1.ReplicaAssignmentBrokers) *v1alpha1.ReplicaAssignmentBrokers {
+	if in == nil {
+		return nil
+	}
+	o := &v1alpha1.ReplicaAssignmentBrokers{
+		Replicas: in.Replicas,
+	}
+	return o
+}
+
+// IsUpToDate checks whether current state is up-to-date compared to the given set of parameters.
+func IsUpToDate(in *v1alpha1.TopicParameters, observed *arv1.TopicDetail) (bool, error) {
+	desired := in.DeepCopy()
+	actual, err := GenerateTopicParameters(observed)
+	if err != nil {
+		return false, err
+	}
+
+	// it says logger isn't found even when logger is imported??
+	// l.Info(cmp.Diff(desired, actual, cmpopts.IgnoreTypes(&runtimev1alpha1.Reference{}, &runtimev1alpha1.Selector{}, []runtimev1alpha1.Reference{})))
+
+	// what is it being compared to, ie which fields should be ignored??
+	return cmp.Equal(desired, actual, cmpopts.EquateEmpty(),
+		cmpopts.IgnoreFields(v1alpha1.TopicParameters{}, "Configs", "KafkaAdminURL", "KafkaAdminURLRef", "KafkaAdminURLSelector"),
+		cmpopts.IgnoreTypes(&runtimev1alpha1.Reference{}, &runtimev1alpha1.Selector{}, []runtimev1alpha1.Reference{})), nil
+}
+
+// GenerateTopicParameters generates *v1alpha1.TopicParameters from *arv1.TopicDetail
+func GenerateTopicParameters(in *arv1.TopicDetail) (*v1alpha1.TopicParameters, error) {
+
+	o := &v1alpha1.TopicParameters{
+		Name: reference.FromPtrValue(in.Name),
+		// if in.Partitions is nil make it 1??
```

Comment:
It should never be nil, but if it was nil I would probably set it to zero ... something to verify when you'll debug the code